### ISSUE TITLE
Lookup plugin should use templates/ folder when used in role

### ISSUE THREAD

**Issue by akloeber** [ansible/ansible#9317]
When using `{{ lookup('template', 'foo.txt') }}` in a role the file is looked up in the `files` folder. To make this compatible with the directory layout for roles the lookup should be based on the `templates` folder instead (maybe the `files` folder can be used as a fallback to keep backwards compatibility).

A minimal example has been provided at https://github.com/akloeber/misc/blob/master/ansible_issue_9317.zip. Its output is:

```
$ ansible-playbook site.yml -i hosts -u deploy && ansible --version

PLAY [all] ******************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [main-test]

TASK: [foo | debug msg=] ****************************************************** 
ok: [main-test] => {
    "msg": "main-test"
}

TASK: [foo | debug msg=] ****************************************************** 
ok: [main-test] => {
    "msg": ""
}

PLAY RECAP ******************************************************************** 
main-test                  : ok=3    changed=0    unreachable=0    failed=0   

ansible 1.7.2
```

Lookup in `files` folder of roles has been added when fixing #2697.

**Comment by bcoca** [comment]
overlaps with https://github.com/ansible/ansible/issues/9551

**Comment by jimi-c** [comment]
This issue has been resolved as of 2.0 and works in current devel/stable-2.1. As such, we will close it now.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
